src/MTProtoTools/Files.php

```diff
@@ -77,15 +77,21 @@ trait Files
      */
     public function wrapMedia(array $media, bool $protected = false): ?Media
     {
+        if ($media['_'] == 'decryptedMessageMediaPhoto') {
+            return new Photo($this, $media, true);
+        }
         if ($media['_'] === 'messageMediaPhoto') {
             if (!isset($media['photo'])) {
                 return null;
             }
             return new Photo($this, $media, $protected);
         }
-        if ($media['_'] !== 'messageMediaDocument') {
+        if ($media['_'] !== 'messageMediaDocument' && $media['_'] !== 'decryptedMessageMediaDocument' && $media['_'] !== 'decryptedMessageMediaExternalDocument') {
```

Comment:
This does not consider all of the media types for all layers: https://core.telegram.org/type/DecryptedMessageMedia

Reply:
its marked as Todo and depend on closing #1415